lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/PoisonTest.cs

```diff
@@ -0,0 +1,135 @@
+using Org.Apache.REEF.Driver.Bridge;
+using Org.Apache.REEF.Driver.Context;
+using Org.Apache.REEF.Driver.Evaluator;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Tang.Implementations.Tang;
+using Org.Apache.REEF.Tang.Interface;
+using Org.Apache.REEF.Tang.Util;
+using Org.Apache.REEF.Utilities.Logging;
+using Xunit;
+using System.Threading;
+using Org.Apache.REEF.Common.Context;
+using Org.Apache.REEF.Common.Events;
+using Org.Apache.REEF.Common.Poison;
+
+namespace Org.Apache.REEF.Tests.Functional.FaultTolerant
+{
+    [Collection("FunctionalTests")]
+    public sealed class PoisonTest : ReefFunctionalTest
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(PoisonTest));
+
+        private const string FailedEvaluatorMessage = "I have succeeded in seeing a failed evaluator.";
+        private const string TaskId = "1234567";
+
+        [Fact]
+        [Trait("Description", "Test Poison functionality by injecting fault in context start handler.")]
+        public void TestPoisonedEvaluatorStartHanlder()
+        {
+            string testFolder = DefaultRuntimeFolder + TestId;
+            TestRun(DriverConfigurations(), typeof(PoisonedEvaluatorDriver), 1, "poisonedEvaluatorStartTest", "local", testFolder);
+            ValidateMessageSuccessfullyLoggedForDriver(FailedEvaluatorMessage, testFolder);
+            CleanUp(testFolder);
+        }
+
+        public IConfiguration DriverConfigurations()
+        {
+            return DriverConfiguration.ConfigurationModule
+                .Set(DriverConfiguration.OnDriverStarted, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnEvaluatorAllocated, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnEvaluatorFailed, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnContextActive, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Build();
+        }
+
+        private sealed class PoisonedEvaluatorDriver :
+            IObserver<IDriverStarted>,
+            IObserver<IAllocatedEvaluator>,
+            IObserver<IActiveContext>,
+            IObserver<IFailedEvaluator>
+        {
+            private readonly IEvaluatorRequestor _requestor;
+
+            [Inject]
+            private PoisonedEvaluatorDriver(IEvaluatorRequestor requestor)
+            {
+                _requestor = requestor;
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().Build());
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                value.SubmitContext(ContextConfiguration.ConfigurationModule
+                        .Set(ContextConfiguration.Identifier, "ContextID")
+                        .Set(ContextConfiguration.OnContextStart, GenericType<PoisonedEventHandler<IContextStart>>.Class)
```

Comment:
As this is to test PoisonedEventHandler itself, you might want to add cases to set named parameters for PoisonedEventHandler in additional to use default values.